src/common/sha2/sha2_c.c

```diff
@@ -633,20 +707,40 @@ void oqs_sha2_sha384_inc_blocks_c(sha384ctx *state, const uint8_t *in, size_t in
 
 void oqs_sha2_sha256_inc_finalize_c(uint8_t *out, sha256ctx *state, const uint8_t *in, size_t inlen) {
 	uint8_t padded[128];
-	uint64_t bytes = load_bigendian_64(state->ctx + 32) + inlen;
 
-	crypto_hashblocks_sha256_c(state->ctx, in, inlen);
-	in += inlen;
-	inlen &= 63;
-	in -= inlen;
+	size_t new_inlen = state->data_len + inlen;
+	size_t tmp_len = new_inlen;
+	const uint8_t *new_in;
+	uint8_t *tmp_in = NULL;
+
+	if (new_inlen == inlen) {
+		new_in = in;
+	} else { //Combine incremental data with final input
+		tmp_in = malloc(tmp_len);
+		size_t i, j;
+		for (i = 0; i < state->data_len; ++i) {
+			tmp_in[i] = state->data[i];
+		}
+		for (j = 0; j < inlen; i++, j++) {
+			tmp_in[i] = in[j];
+		}
```

Comment:
The benefit of `memcpy` at compile time it enables optimization (mainly alignment and block size). So instead of adding attribute to align `state->data` and `tmp_in` (8 bytes or 4 bytes depend on platform), compiler helps us out in this case. 

Other than that, compiler has knowledge about the system `-mtune=native` or `-march=native` so it can adjust the blocksize more efficiently (as for now, the for loop blocksize is 1 byte). 

https://www.embedded.com/optimizing-memcpy-improves-speed/

Reply:
I was attempting to stay consistent with the look and layout of the original code.